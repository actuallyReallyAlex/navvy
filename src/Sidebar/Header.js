import React from 'react'
import PropTypes from 'prop-types'
import { Text } from 'grommet'

const Header = ({ heading, size }) => {
  return (
    <Text
      margin={{ horizontal: 'small', top: 'medium' }}
      size="small"
      style={{ textAlign: size === 'expanded' ? 'start' : 'center' }}
    >
      {heading}
    </Text>
  )
}

Header.propTypes = {
  heading: PropTypes.string.isRequired,
  size: PropTypes.oneOf(['expanded', 'collapsed']).isRequired
}

export default Header
