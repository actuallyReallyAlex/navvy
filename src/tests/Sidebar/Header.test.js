import React from 'react'
import Header from '../../Sidebar/Header'
import { shallow } from 'enzyme'

const defaultProps = {
  heading: 'Heading',
  size: 'expanded'
}

describe('<Header />', () => {
  afterEach(() => {
    // defaultProps.handler.mockClear()
  })

  test('Should render the <Header /> component.', () => {
    const component = shallow(<Header {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
