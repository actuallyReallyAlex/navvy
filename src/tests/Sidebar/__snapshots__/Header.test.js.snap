// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render the <Header /> component with sidebar collapsed. 1`] = `ShallowWrapper {}`;

exports[`<Header /> Should render the <Header /> component with sidebar expanded. 1`] = `ShallowWrapper {}`;
